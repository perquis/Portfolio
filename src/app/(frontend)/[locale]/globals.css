@tailwind base;
@tailwind components;
@tailwind utilities;
@import url(../../../styles/input.css);
@import url(../../../styles/themes.css);

:root {
  --tw-white: hsl(0, 0%, 100%);
  --tw-zinc-50: hsl(240, 5%, 96%);
  --tw-zinc-900: hsl(240, 4%, 16%);
  --tw-zinc-950: hsl(240, 6%, 10%);
  --tw-indigo-600: hsla(243, 75%, 59%, 0.75);
}

*:focus-visible {
  outline: 2px solid var(--tw-indigo-600) !important;
  outline-offset: 3px !important;
  box-sizing: border-box;
}

a {
  @apply rounded-sm;
}

::-moz-selection {
  color: var(--tw-white);
  background: var(--tw-indigo-600);
}

::selection {
  color: var(--tw-white);
  background: var(--tw-indigo-600);
}

html {
  scroll-behavior: smooth !important;
  scroll-padding-top: 64px;
}

.sandpack * {
  outline: none !important;
}

@layer base {
  :root {
    --background: 0 0% 100%;
    --foreground: 240 10% 3.9%;
    --card: 0 0% 100%;
    --card-foreground: 240 10% 3.9%;
    --popover: 0 0% 100%;
    --popover-foreground: 240 10% 3.9%;
    --primary: 240 5.9% 10%;
    --primary-foreground: 0 0% 98%;
    --secondary: 240 4.8% 95.9%;
    --secondary-foreground: 240 5.9% 10%;
    --muted: 240 4.8% 95.9%;
    --muted-foreground: 240 3.8% 46.1%;
    --accent: 240 4.8% 95.9%;
    --accent-foreground: 240 5.9% 10%;
    --destructive: 0 84.2% 60.2%;
    --destructive-foreground: 0 0% 98%;
    --border: 240 5.9% 90%;
    --input: 240 5.9% 90%;
    --ring: 240 10% 3.9%;
    --chart-1: 12 76% 61%;
    --chart-2: 173 58% 39%;
    --chart-3: 197 37% 24%;
    --chart-4: 43 74% 66%;
    --chart-5: 27 87% 67%;
    --radius: 0.5rem;
    --color-1: 0 100% 63%;
    --color-2: 270 100% 63%;
    --color-3: 210 100% 63%;
    --color-4: 195 100% 63%;
    --color-5: 90 100% 63%;
  }

  [data-theme="dark"] {
    --background: 240 10% 3.9%;
    --foreground: 0 0% 98%;
    --card: 240 10% 3.9%;
    --card-foreground: 0 0% 98%;
    --popover: 240 10% 3.9%;
    --popover-foreground: 0 0% 98%;
    --primary: 0 0% 98%;
    --primary-foreground: 240 5.9% 10%;
    --secondary: 240 3.7% 15.9%;
    --secondary-foreground: 0 0% 98%;
    --muted: 240 3.7% 15.9%;
    --muted-foreground: 240 5% 64.9%;
    --accent: 240 6% 10%;
    --accent-foreground: 0 0% 98%;
    --destructive: 0 62.8% 30.6%;
    --destructive-foreground: 0 0% 98%;
    --border: 240 3.7% 15.9%;
    --input: 240 3.7% 15.9%;
    --ring: 240 4.9% 83.9%;
    --chart-1: 220 70% 50%;
    --chart-2: 160 60% 45%;
    --chart-3: 30 80% 55%;
    --chart-4: 280 65% 60%;
    --chart-5: 340 75% 55%;
    --color-1: 0 100% 63%;
    --color-2: 270 100% 63%;
    --color-3: 210 100% 63%;
    --color-4: 195 100% 63%;
    --color-5: 90 100% 63%;
  }
}

@layer base {
  * {
    @apply border-border;
  }
  body {
    @apply bg-background text-foreground;
  }
}
