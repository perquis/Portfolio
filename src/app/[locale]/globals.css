@tailwind base;
@tailwind components;
@tailwind utilities;

:root {
  --tw-white: hsl(0, 0%, 100%);
  --tw-zinc-50: hsl(240, 5%, 96%);
  --tw-zinc-900: hsl(240, 4%, 16%);
  --tw-zinc-950: hsl(240, 6%, 10%);
  --tw-indigo-600: hsla(243, 75%, 59%, 0.75);
}

*:focus-visible {
  outline: 2px solid var(--tw-indigo-600) !important;
  outline-offset: 3px !important;
  box-sizing: border-box;
}

::-moz-selection {
  color: var(--tw-white);
  background: var(--tw-indigo-600);
}

::selection {
  color: var(--tw-white);
  background: var(--tw-indigo-600);
}

@import url(../../styles/input.css);
@import url(../../styles/themes.css);

.sandpack * {
  outline: none !important;
}
