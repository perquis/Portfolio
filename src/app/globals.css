@tailwind base;
@tailwind components;
@tailwind utilities;

:root {
  --tw-indigo-600: hsla(243, 75%, 59%, 0.75);
  --tw-rose-600: hsla(340, 82%, 55%, 0.75);
  --tw-white: hsl(0, 0%, 100%);
}

*:focus-visible {
  outline: 2px solid var(--tw-indigo-600) !important;
  outline-offset: 3px !important;
  box-sizing: border-box;
}

::-moz-selection {
  color: var(--tw-white);
  background: var(--tw-indigo-600);
}

::selection {
  color: var(--tw-white);
  background: var(--tw-indigo-600);
}

html[data-mode="dark"] *:focus-visible {
  outline: 2px solid var(--tw-rose-600) !important;
}

html[data-mode="dark"] ::-moz-selection {
  background: var(--tw-rose-600);
}

html[data-mode="dark"] ::selection {
  background: var(--tw-rose-600);
}
