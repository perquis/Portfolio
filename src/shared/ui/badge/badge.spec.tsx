import { expect, it } from "@jest/globals";
import renderer from "react-test-renderer";

import type { Color } from "@/interfaces/tailwindcss";
import type { Rounded } from "@/interfaces/variants";

import Badge from "./badge";

interface BadgeTestInput {
  color: Color;
  rounded: Rounded;
}

describe("Badge", () => {
  it.each([
    {
      color: "amber",
      rounded: "default",
    },
    {
      color: "blue",
      rounded: "default",
    },
    {
      color: "cyan",
      rounded: "default",
    },
    {
      color: "emerald",
      rounded: "default",
    },
    {
      color: "fuchsia",
      rounded: "default",
    },
    {
      color: "green",
      rounded: "default",
    },
    {
      color: "indigo",
      rounded: "default",
    },
    {
      color: "lime",
      rounded: "default",
    },
    {
      color: "neutral",
      rounded: "default",
    },
    {
      color: "orange",
      rounded: "default",
    },
    {
      color: "pink",
      rounded: "default",
    },
    {
      color: "purple",
      rounded: "default",
    },
    {
      color: "red",
      rounded: "default",
    },
    {
      color: "rose",
      rounded: "default",
    },
    {
      color: "slate",
      rounded: "default",
    },
    {
      color: "stone",
      rounded: "default",
    },
    {
      color: "teal",
      rounded: "default",
    },
    {
      color: "violet",
      rounded: "default",
    },
    {
      color: "yellow",
      rounded: "default",
    },
    {
      color: "zinc",
      rounded: "default",
    },
    {
      color: "amber",
      rounded: "full",
    },
    {
      color: "blue",
      rounded: "full",
    },
    {
      color: "cyan",
      rounded: "full",
    },
    {
      color: "emerald",
      rounded: "full",
    },
    {
      color: "fuchsia",
      rounded: "full",
    },
    {
      color: "green",
      rounded: "full",
    },
    {
      color: "indigo",
      rounded: "full",
    },
    {
      color: "lime",
      rounded: "full",
    },
    {
      color: "neutral",
      rounded: "full",
    },
    {
      color: "orange",
      rounded: "full",
    },
    {
      color: "pink",
      rounded: "full",
    },
    {
      color: "purple",
      rounded: "full",
    },
    {
      color: "red",
      rounded: "full",
    },
    {
      color: "rose",
      rounded: "full",
    },
    {
      color: "slate",
      rounded: "full",
    },
    {
      color: "stone",
      rounded: "full",
    },
    {
      color: "teal",
      rounded: "full",
    },
    {
      color: "violet",
      rounded: "full",
    },
    {
      color: "yellow",
      rounded: "full",
    },
    {
      color: "zinc",
      rounded: "full",
    },
  ] as BadgeTestInput[])("renders correctly for color $color and rounded $rounded", ({ color, rounded }) => {
    const tree = renderer.create(<Badge color={color} rounded={rounded} />).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
