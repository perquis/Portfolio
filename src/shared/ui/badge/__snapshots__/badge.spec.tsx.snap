// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Badge renders correctly for color amber and rounded default 1`] = `
<span
  className="px-3 py-2 text-sm rounded-md bg-amber-100 dark:bg-amber-950 border-amber-300 dark:border-amber-700 text-amber-600 dark:text-amber-400"
/>
`;

exports[`Badge renders correctly for color amber and rounded full 1`] = `
<span
  className="px-3 py-2 text-sm rounded-full bg-amber-100 dark:bg-amber-950 border-amber-300 dark:border-amber-700 text-amber-600 dark:text-amber-400"
/>
`;

exports[`Badge renders correctly for color blue and rounded default 1`] = `
<span
  className="px-3 py-2 text-sm rounded-md bg-blue-100 dark:bg-blue-950 border-blue-300 dark:border-blue-700 text-blue-600 dark:text-blue-400"
/>
`;

exports[`Badge renders correctly for color blue and rounded full 1`] = `
<span
  className="px-3 py-2 text-sm rounded-full bg-blue-100 dark:bg-blue-950 border-blue-300 dark:border-blue-700 text-blue-600 dark:text-blue-400"
/>
`;

exports[`Badge renders correctly for color cyan and rounded default 1`] = `
<span
  className="px-3 py-2 text-sm rounded-md bg-cyan-100 dark:bg-cyan-950 border-cyan-300 dark:border-cyan-700 text-cyan-600 dark:text-cyan-400"
/>
`;

exports[`Badge renders correctly for color cyan and rounded full 1`] = `
<span
  className="px-3 py-2 text-sm rounded-full bg-cyan-100 dark:bg-cyan-950 border-cyan-300 dark:border-cyan-700 text-cyan-600 dark:text-cyan-400"
/>
`;

exports[`Badge renders correctly for color emerald and rounded default 1`] = `
<span
  className="px-3 py-2 text-sm rounded-md bg-emerald-100 dark:bg-emerald-950 border-emerald-300 dark:border-emerald-700 text-emerald-600 dark:text-emerald-400"
/>
`;

exports[`Badge renders correctly for color emerald and rounded full 1`] = `
<span
  className="px-3 py-2 text-sm rounded-full bg-emerald-100 dark:bg-emerald-950 border-emerald-300 dark:border-emerald-700 text-emerald-600 dark:text-emerald-400"
/>
`;

exports[`Badge renders correctly for color fuchsia and rounded default 1`] = `
<span
  className="px-3 py-2 text-sm rounded-md bg-fuchsia-100 dark:bg-fuchsia-950 border-fuchsia-300 dark:border-fuchsia-700 text-fuchsia-600 dark:text-fuchsia-400"
/>
`;

exports[`Badge renders correctly for color fuchsia and rounded full 1`] = `
<span
  className="px-3 py-2 text-sm rounded-full bg-fuchsia-100 dark:bg-fuchsia-950 border-fuchsia-300 dark:border-fuchsia-700 text-fuchsia-600 dark:text-fuchsia-400"
/>
`;

exports[`Badge renders correctly for color green and rounded default 1`] = `
<span
  className="px-3 py-2 text-sm rounded-md bg-green-100 dark:bg-green-950 border-green-300 dark:border-green-700 text-green-600 dark:text-green-400"
/>
`;

exports[`Badge renders correctly for color green and rounded full 1`] = `
<span
  className="px-3 py-2 text-sm rounded-full bg-green-100 dark:bg-green-950 border-green-300 dark:border-green-700 text-green-600 dark:text-green-400"
/>
`;

exports[`Badge renders correctly for color indigo and rounded default 1`] = `
<span
  className="px-3 py-2 text-sm rounded-md bg-indigo-100 dark:bg-indigo-950 border-indigo-300 dark:border-indigo-700 text-indigo-600 dark:text-indigo-400"
/>
`;

exports[`Badge renders correctly for color indigo and rounded full 1`] = `
<span
  className="px-3 py-2 text-sm rounded-full bg-indigo-100 dark:bg-indigo-950 border-indigo-300 dark:border-indigo-700 text-indigo-600 dark:text-indigo-400"
/>
`;

exports[`Badge renders correctly for color lime and rounded default 1`] = `
<span
  className="px-3 py-2 text-sm rounded-md bg-lime-100 dark:bg-lime-950 border-lime-300 dark:border-lime-700 text-lime-600 dark:text-lime-400"
/>
`;

exports[`Badge renders correctly for color lime and rounded full 1`] = `
<span
  className="px-3 py-2 text-sm rounded-full bg-lime-100 dark:bg-lime-950 border-lime-300 dark:border-lime-700 text-lime-600 dark:text-lime-400"
/>
`;

exports[`Badge renders correctly for color neutral and rounded default 1`] = `
<span
  className="px-3 py-2 text-sm rounded-md bg-neutral-100 dark:bg-neutral-950 border-neutral-300 dark:border-neutral-700 text-neutral-600 dark:text-neutral-400"
/>
`;

exports[`Badge renders correctly for color neutral and rounded full 1`] = `
<span
  className="px-3 py-2 text-sm rounded-full bg-neutral-100 dark:bg-neutral-950 border-neutral-300 dark:border-neutral-700 text-neutral-600 dark:text-neutral-400"
/>
`;

exports[`Badge renders correctly for color orange and rounded default 1`] = `
<span
  className="px-3 py-2 text-sm rounded-md bg-orange-100 dark:bg-orange-950 border-orange-300 dark:border-orange-700 text-orange-600 dark:text-orange-400"
/>
`;

exports[`Badge renders correctly for color orange and rounded full 1`] = `
<span
  className="px-3 py-2 text-sm rounded-full bg-orange-100 dark:bg-orange-950 border-orange-300 dark:border-orange-700 text-orange-600 dark:text-orange-400"
/>
`;

exports[`Badge renders correctly for color pink and rounded default 1`] = `
<span
  className="px-3 py-2 text-sm rounded-md bg-pink-100 dark:bg-pink-950 border-pink-300 dark:border-pink-700 text-pink-600 dark:text-pink-400"
/>
`;

exports[`Badge renders correctly for color pink and rounded full 1`] = `
<span
  className="px-3 py-2 text-sm rounded-full bg-pink-100 dark:bg-pink-950 border-pink-300 dark:border-pink-700 text-pink-600 dark:text-pink-400"
/>
`;

exports[`Badge renders correctly for color purple and rounded default 1`] = `
<span
  className="px-3 py-2 text-sm rounded-md bg-purple-100 dark:bg-purple-950 border-purple-300 dark:border-purple-700 text-purple-600 dark:text-purple-400"
/>
`;

exports[`Badge renders correctly for color purple and rounded full 1`] = `
<span
  className="px-3 py-2 text-sm rounded-full bg-purple-100 dark:bg-purple-950 border-purple-300 dark:border-purple-700 text-purple-600 dark:text-purple-400"
/>
`;

exports[`Badge renders correctly for color red and rounded default 1`] = `
<span
  className="px-3 py-2 text-sm rounded-md bg-red-100 dark:bg-red-950 border-red-300 dark:border-red-700 text-red-600 dark:text-red-400"
/>
`;

exports[`Badge renders correctly for color red and rounded full 1`] = `
<span
  className="px-3 py-2 text-sm rounded-full bg-red-100 dark:bg-red-950 border-red-300 dark:border-red-700 text-red-600 dark:text-red-400"
/>
`;

exports[`Badge renders correctly for color rose and rounded default 1`] = `
<span
  className="px-3 py-2 text-sm rounded-md bg-rose-100 dark:bg-rose-950 border-rose-300 dark:border-rose-700 text-rose-600 dark:text-rose-400"
/>
`;

exports[`Badge renders correctly for color rose and rounded full 1`] = `
<span
  className="px-3 py-2 text-sm rounded-full bg-rose-100 dark:bg-rose-950 border-rose-300 dark:border-rose-700 text-rose-600 dark:text-rose-400"
/>
`;

exports[`Badge renders correctly for color slate and rounded default 1`] = `
<span
  className="px-3 py-2 text-sm rounded-md bg-slate-100 dark:bg-slate-950 border-slate-300 dark:border-slate-700 text-slate-600 dark:text-slate-400"
/>
`;

exports[`Badge renders correctly for color slate and rounded full 1`] = `
<span
  className="px-3 py-2 text-sm rounded-full bg-slate-100 dark:bg-slate-950 border-slate-300 dark:border-slate-700 text-slate-600 dark:text-slate-400"
/>
`;

exports[`Badge renders correctly for color stone and rounded default 1`] = `
<span
  className="px-3 py-2 text-sm rounded-md bg-stone-100 dark:bg-stone-950 border-stone-300 dark:border-stone-700 text-stone-600 dark:text-stone-400"
/>
`;

exports[`Badge renders correctly for color stone and rounded full 1`] = `
<span
  className="px-3 py-2 text-sm rounded-full bg-stone-100 dark:bg-stone-950 border-stone-300 dark:border-stone-700 text-stone-600 dark:text-stone-400"
/>
`;

exports[`Badge renders correctly for color teal and rounded default 1`] = `
<span
  className="px-3 py-2 text-sm rounded-md bg-teal-100 dark:bg-teal-950 border-teal-300 dark:border-teal-700 text-teal-600 dark:text-teal-400"
/>
`;

exports[`Badge renders correctly for color teal and rounded full 1`] = `
<span
  className="px-3 py-2 text-sm rounded-full bg-teal-100 dark:bg-teal-950 border-teal-300 dark:border-teal-700 text-teal-600 dark:text-teal-400"
/>
`;

exports[`Badge renders correctly for color violet and rounded default 1`] = `
<span
  className="px-3 py-2 text-sm rounded-md bg-violet-100 dark:bg-violet-950 border-violet-300 dark:border-violet-700 text-violet-600 dark:text-violet-400"
/>
`;

exports[`Badge renders correctly for color violet and rounded full 1`] = `
<span
  className="px-3 py-2 text-sm rounded-full bg-violet-100 dark:bg-violet-950 border-violet-300 dark:border-violet-700 text-violet-600 dark:text-violet-400"
/>
`;

exports[`Badge renders correctly for color yellow and rounded default 1`] = `
<span
  className="px-3 py-2 text-sm rounded-md bg-yellow-100 dark:bg-yellow-950 border-yellow-300 dark:border-yellow-700 text-yellow-600 dark:text-yellow-400"
/>
`;

exports[`Badge renders correctly for color yellow and rounded full 1`] = `
<span
  className="px-3 py-2 text-sm rounded-full bg-yellow-100 dark:bg-yellow-950 border-yellow-300 dark:border-yellow-700 text-yellow-600 dark:text-yellow-400"
/>
`;

exports[`Badge renders correctly for color zinc and rounded default 1`] = `
<span
  className="px-3 py-2 text-sm rounded-md bg-zinc-100 dark:bg-zinc-900 border-zinc-300 dark:border-zinc-700 text-zinc-600 dark:text-zinc-400"
/>
`;

exports[`Badge renders correctly for color zinc and rounded full 1`] = `
<span
  className="px-3 py-2 text-sm rounded-full bg-zinc-100 dark:bg-zinc-900 border-zinc-300 dark:border-zinc-700 text-zinc-600 dark:text-zinc-400"
/>
`;
